src/Raven.Client/Json/Converters/JsonDictionaryDateTimeKeysConverter.cs

```diff
@@ -121,7 +123,12 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 
         public override bool CanConvert(Type objectType)
         {
-            if (objectType.GetTypeInfo().IsGenericType == false)
+            return _cache.GetOrAdd(objectType, CanConvertInternal);
```

Comment:
This is expensive?

Reply:
Original implementation took 1600ms in my test. This one takes 400ms.